docs/enhancements/multi-rhel-fips.md

```diff
@@ -0,0 +1,244 @@
+  clusters using installer binaries built against RHEL 8 or RHEL 9
+
+- Allow for FIPS compliant clusters to be installed from the SaaS offering or
+  the on-prem offering
+
+### Non-Goals
+
+- Changing cluster install interfaces to accommodate new FIPS requirements
+  should be avoided
+
+- Dynamically determining a given release's RHEL version. Assisted service will
+  track the minimum version for using el9 and if a version can't be determined
+  for some reason (FCOS may not have the same versioning scheme) el9 will be
+  the default.
+
+## Proposal
+
+Two additional containers will run alongside assisted-service in the same pod.
+These "installer-runner" containers will expose a HTTP API local to the pod
+using a unix socket. assisted-service can then choose which API to contact to
+run an installer binary for a specified release to generate the manifests
+required for a particular install. These manifests will then be uploaded to
+whatever storage is in use for this deployment (local for on-prem, or s3 for
+SaaS) and assisted-service will take over as usual from there.
+
+### User Stories
+
+#### Story 1
+
+As a Cluster Creator, I want to install FIPS compliant clusters for any supported OpenShift version
+
+### Implementation Details/Notes/Constraints
+
+#### New Images
+
+Two new container images will need to be built, and published for every release
+footprint we support. These images will be created based on existing
+assisted-service code, but could be split into their own independent projects
+later.
+
+### Risks and Mitigations
+
+Shipping a new image is a non-trivial process. This may take more time than we
+have to set up. We could likely get away with using the existing
+assisted-service image with a different entrypoint for one of the runner images,
+but that still requires us to publish a new one for the architecture
+assisted-service will not be using.
+
+## Design Details [optional]
+
+- A new `installer-runner` service will be created, written in go.
+- The installer-runner will be compiled twice: once in a RHEL 8 builder image,
+  and once in a RHEL 9 builder image, with each resulting binary being placed
+  into a RHEL base image of corresponding version.
+
+
+The new runner containers will expose a HTTP server using a unix socket.
+assisted-service will POST to one of these servers when it needs manifests generated.
+The runner container will respond with any error that occurred while generating
+the manifests or with success in which case assisted-service will assume the
+manifests were created and uploaded successfully.
+
+### API
+
+The new services would effectively wrap the existing `InstallConfigGenerator`
+interface.
+
+API call input:
+- common.Cluster json
+- install config
+- release image
+
+API call output:
+- Appropriate HTTP response
+- Error message if the call was not successful
+
+### Installer Cache
+
+The installer cache directory will be shared (as it's currently on the PV), but
```

Comment:
"as it's currently on the PV" <- Do you mean that the installer cache is on a PV in non SAAS environments.
As far as I am aware, SAAS does not use a PV.
Could you clarify what you mean by this?